src/bp_sim.cpp

```diff
@@ -3808,10 +3808,8 @@ void CNodeGenerator::handle_pcap_pkt(CGenNode *node, CFlowGenListPerThread *thre
 
 void CNodeGenerator::handle_timesync_msg(CGenNodeTimesync *node, CFlowGenListPerThread *thread, bool &exit_scheduler) {
     m_p_queue.pop();
-    if ((m_p_queue.size() == m_parent->m_non_active_nodes) ||  // exit in case this is the last node
-        (node->is_mask_for_free())) {                          // exit in case of stopped traffic (Tx side)
+    if (node->is_mask_for_free()) {
         thread->free_node((CGenNode *)node);
-        exit_scheduler = true;
```

Comment:
Why do you find this line irrelevant?

Reply:
See my comment above

Reply:
Hmm... In this case I should also remove exit_scheduler from function definition.
I'll do this tommorow